app.py

```diff
@@ -39,6 +46,50 @@ def init_app(self):
                                      {"type": "panel", 
                                       "short_name": "shotgun_panel"})
         
+    def navigate(self, entity_type, entity_id, mode):
+        """
+        API support to start the panel and navigate to a location.
+        
+        Depending on the mode parameter, the window behavior may 
+        differ, but the general idea is that if a panel window doesn't exist
+        or isn't in focus, it is created and/or brought to front before navigating
+        to the given entity. The new location is added to the existing history stack,
+        allowing the user to easily move back again if needed and effectively undo 
+        the programmatic navigation.
+        
+        The following modes exist:
+        - PANEL - launch the panel as a panel. Panels are always singleton, so this will start
+          the panel if it doesn't already exist. If it exists it will be given user focus. Note that
+          on engines without panel support implemented, this flag will be equivalent to DIALOG
+          below.
+        - DIALOG - launch the panel as a dialog. A dialog may co-exist with a panel. if a dialog
+          already exists, this is given focus.
```

Comment:
Minor:
- How about "it" is given focus? Otherwise "this" could be ambiguous.
- Capital "I" on "if a dialog[...]"?